src/d/lyt/msg_window/d_lyt_simple_window.cpp

```diff
@@ -0,0 +1,796 @@
+    mLyt.calc();
+    mpPanes[SIMPLE_WINDOW_SWORD_PANE_IN_OUT]->SetVisible(false);
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_COLOR].setAnimEnable(false);
+
+    nw4r::math::MTX34 transform = mpPanes[SIMPLE_WINDOW_SWORD_PANE_A_BTN]->GetGlobalMtx();
+    mVec2_c _unused;
+    _unused.x = transform._03;
+    _unused.y = transform._13;
+
+    mOrigPosition.copyFrom(mpPanes[SIMPLE_WINDOW_SWORD_PANE_POSITION]->GetTranslate());
+    mPosition.copyFrom(mpPanes[SIMPLE_WINDOW_SWORD_PANE_POSITION]->GetTranslate());
+    mOffset.x = mOffset.y = 0.0f;
+
+    field_0x5AC8 = 0;
+
+    return true;
+}
+
+bool dLytSimpleWindowSword_c::remove() {
+    for (int i = 0; i < SIMPLE_WINDOW_SWORD_NUM_ANIMS; i++) {
+        mAnm[i].remove();
+    }
+    return true;
+}
+
+bool dLytSimpleWindowSword_c::execute() {
+    for (int i = 0; i < SIMPLE_WINDOW_SWORD_NUM_ANIMS; i++) {
+        if (mAnm[i].isEnabled()) {
+            mAnm[i].play();
+        }
+    }
+    mLyt.calc();
+    field_0x5AC8 = 0;
+    return true;
+}
+
+bool dLytSimpleWindowSword_c::draw() {
+    mLyt.draw();
+    return true;
+}
+
+void dLytSimpleWindowSword_c::initializeIn(s32 pos) {
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_IN].setAnimEnable(true);
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_IN].setFrame(0.0f);
+
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_POSITION].setFrame(pos);
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_POSITION].setAnimEnable(true);
+    mLyt.calc();
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_POSITION].setAnimEnable(false);
+
+    mPosition = mOrigPosition;
+    mPosition.y = mpPanes[SIMPLE_WINDOW_SWORD_PANE_POSITION]->GetTranslate().y;
+    mpPanes[SIMPLE_WINDOW_SWORD_PANE_POSITION]->SetTranslate(mPosition);
+}
+
+bool dLytSimpleWindowSword_c::isIn() {
+    if (mAnm[SIMPLE_WINDOW_SWORD_ANIM_IN].isEndReached()) {
+        mAnm[SIMPLE_WINDOW_SWORD_ANIM_IN].setAnimEnable(false);
+        return true;
+    }
+    return false;
+}
+
+void dLytSimpleWindowSword_c::initializeOutputText() {
+    nw4r::lyt::Size size = getTextBox()->GetSize();
+    nw4r::lyt::Size fontSize = getTextBox()->GetFontSize();
+    nw4r::math::MTX34 transform = getTextBox()->GetGlobalMtx();
+    mOffset.x = transform._03 - size.width / 2;
+    mOffset.y = transform._13 + size.height / 2;
+}
+
+void dLytSimpleWindowSword_c::initializeOut() {
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_OUT].setAnimEnable(true);
+    mAnm[SIMPLE_WINDOW_SWORD_ANIM_OUT].setFrame(0.0f);
+}
+
+bool dLytSimpleWindowSword_c::isOut() {
+    if (mAnm[SIMPLE_WINDOW_SWORD_ANIM_OUT].isEndReached()) {
+        mAnm[SIMPLE_WINDOW_SWORD_ANIM_OUT].setAnimEnable(false);
+        return true;
+    }
+    return false;
+}
+
+bool dLytSimpleWindowSword_c::setText(const wchar_t *text) {
+    int foundIdx = ARRAY_LENGTH(mCharData);
+    for (int i = 0; i < (int)ARRAY_LENGTH(mCharData); i++) {
+        if (mCharData[i].displayTimerMaybe >= 0 && foundIdx > mCharData[i].field_0x0C) {
+            foundIdx = mCharData[i].field_0x0C;
+        }
+    }
+
+    // How is text bounds-checked here????
```

Comment:
My assumption is because the two buffers used were the same size so it would always be ok in this case?

Reply:
From what I saw called `setText` it would pass a 1023 len buffer always - Did I miss a case?